gradio/utils.py

```diff
@@ -286,13 +286,19 @@ def is_zero_gpu_space() -> bool:
     return os.getenv("SPACES_ZERO_GPU") == "true"
 
 
-def readme_to_html(article: str) -> str:
+def download_if_url(article: str) -> str:
     try:
+        result = urllib.parse.urlparse(article)
+        is_url = all([result.scheme, result.netloc, result.path])
+        is_url = is_url and result.scheme in ["http", "https"]
+    except:
+        is_url = False
+
+    if is_url:
         response = httpx.get(article, timeout=3)
         if response.status_code == httpx.codes.OK:  # pylint: disable=no-member
             article = response.text
-    except (httpx.InvalidURL, httpx.RequestError):
```

Comment:
I think we should still keep the request error exception catching but otherwise this looks good to me. Thanks @YuryYakhno !

Reply:
I think this is fine because @YuryYakhno has moved the exception handling up and made it more generic

Reply:
Yes and if there is an exception, we can just return the original URL string as before